contrib/download-frozen-image-v2.sh

```diff
@@ -201,6 +201,58 @@ handle_single_manifest_v2() {
 	manifestJsonEntries=("${manifestJsonEntries[@]}" "$manifestJsonEntry")
 }
 
+get_target_arch() {
+	if [ -n "${TARGETARCH}" ]; then
+		echo "${TARGETARCH}"
+		return 0
+	fi
+
+	if type go > /dev/null; then
+		echo "$(go env GOARCH)"
+		return 0
+	fi
+
+	if type dpkg > /dev/null; then
+		debArch="$(dpkg --print-architecture)"
+		case "${debArch}" in
+			armel | armhf)
+				echo "arm"
+				return 0
+				;;
+			*)
+				echo "${debArch}"
+				return 0
+				;;
+		esac
+	fi
+
+	if type uname > /dev/null; then
+		uArch="$(uname -m)"
+		case "${uArch}" in
+			x86_64)
+				echo amd64
+				return 0
+				;;
+			arm | armv6l | armv7l)
+				echo arm
+				return 0
+				;;
+			aarch64 | armv8l)
```

Comment:
If we're getting `armv8l`, we're on a 32bit chip (or using `linux32` / `setarch` to pretend), so this should probably just be `arm*` or something similar like the following in the above block instead:

```suggestion
			arm | armv[0-9]*)
				echo arm
				return 0
				;;
			aarch64)
```